app/controllers/concerns/admin_only_access.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# AdminOnlyAccess provides a mechanism to restrict access to certain
+# controller actions to admin users only. It hooks into the controller's
+# lifecycle using a before_action to check the user's admin status.
+module AdminOnlyAccess
```

Comment:
For any page that requires a user to be an admin in order to access, if the user is not an admin we are redirecting them to a 404 page for security reasons.